crates/ruff_db/src/system/os.rs

```diff
@@ -81,3 +89,74 @@ impl From<std::fs::FileType> for FileType {
         }
     }
 }
+
+impl TryFrom<camino::Utf8DirEntry> for DirEntry {
+    type Error = std::io::Error;
+
+    fn try_from(value: camino::Utf8DirEntry) -> Result<Self> {
+        Ok(Self {
+            path: SystemPathBuf::from_utf8_path_buf(value.path().to_path_buf()),
+            file_type: FileType::from(value.file_type()?),
```

Comment:
Interesting, why can file_type fail. Should our DirEntry also implement file_type lazily (by storing a Result<FileType>?)

Reply:
> Interesting, why can file_type fail.

Unclear. The `camino::Utf8DirEntry::file_type` method just delegates to `std::fs::DirEntry::file_type`. Neither method's documentation gives any information about why it might return an error.

> Should our DirEntry also implement file_type lazily (by storing a Result?)

Sure, I can do that.